test/integration/connectors/test_s3.py

```diff
@@ -71,6 +71,29 @@ async def test_s3_source(anon_connection_config: S3ConnectionConfig):
         )
 
 
+@pytest.mark.asyncio
+@pytest.mark.tags(CONNECTOR_TYPE, SOURCE_TAG)
+async def test_s3_source_special_char(anon_connection_config: S3ConnectionConfig):
+    indexer_config = S3IndexerConfig(remote_url="s3://utic-dev-tech-fixtures/special-characters/")
+    with tempfile.TemporaryDirectory() as tempdir:
+        tempdir_path = Path(tempdir)
+        download_config = S3DownloaderConfig(download_dir=tempdir_path)
+        indexer = S3Indexer(connection_config=anon_connection_config, index_config=indexer_config)
+        downloader = S3Downloader(
+            connection_config=anon_connection_config, download_config=download_config
+        )
+        await source_connector_validation(
+            indexer=indexer,
+            downloader=downloader,
+            configs=ValidationConfigs(
+                test_id="s3-specialchar",
+                predownload_file_data_check=validate_predownload_file_data,
+                postdownload_file_data_check=validate_postdownload_file_data,
+                expected_num_files=1,
+            ),
+        )
+
+
```

Comment:
This validation code extracts only files (not directories) from the download directory, then asserts they match the expected directory structure:

* https://github.com/Unstructured-IO/unstructured-ingest/blob/ca90f0de698108171904b71abb628fbdf2cea2cb/test/integration/connectors/utils/validation.py#L287-L291
* https://github.com/Unstructured-IO/unstructured-ingest/blob/ca90f0de698108171904b71abb628fbdf2cea2cb/test/integration/connectors/utils/validation.py#L127